### PR TITLE
Cfy 5135 ssl on file server

### PR DESCRIPTION
**Pull request by noakup**

### CODE REVIEW

File: components/manager/scripts/set_manager_ips.py [hunk 1]
```diff
@@ -82,3 +86,17 @@
                               format(rest_host_external_endpoint_type))
 ctx.logger.info('external_rest_host set to: {0}'.format(
     source_runtime_props['external_rest_host']))
+
+# set default resolving rules based on security configuration
+if source_runtime_props['file_server_protocol'] == 'https':
```
Comment (dankilman) [inline components/manager/scripts/set_manager_ips.py]: remove

File: components/mgmtworker/scripts/preconfigure.py [hunk 2]
```diff
@@ -50,8 +48,35 @@
...
 
-source_runtime_props['file_server_host'] = file_server_host
+# the file server is accessed through the same host and protocol as the rest
+# service, but on a different port
+file_server_host = target_runtime_props['file_server_host']
+file_server_port = target_runtime_props['file_server_port']
+file_server_protocol = target_runtime_props['file_server_protocol']
+
+
+ctx.logger.info('mgmtworker rest_host: {0}'.format(rest_host))
```
Comment (dankilman) [inline components/mgmtworker/scripts/preconfigure.py]: removed

File: types/manager-types.yaml [hunk 3]
```diff
@@ -800,6 +806,12 @@ node_types:
             script_path:
               default: components/mgmtworker/scripts/configure.py
             fabric_env: *simple_fabric_env
+        configure:
```
Comment (dankilman) [inline types/manager-types.yaml]: remove

File: types/manager-types.yaml [hunk 4]
```diff
@@ -583,6 +583,12 @@ node_types:
             script_path:
               default: components/riemann/scripts/configure.py
             fabric_env: *simple_fabric_env
+        configure:
```
Comment (dankilman) [inline types/manager-types.yaml]: remove

File: components/nginx/config/http-file-server.cloudify [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# File server.
+server {
+    # server listening
+    listen              *:53229;
```
Comment (dankilman) [inline components/nginx/config/http-file-server.cloudify]: remove or add?

File: components/nginx/config/default.conf [hunk 6]
```diff
@@ -24,15 +24,4 @@ upstream cloudify-resources {
...
-
-  access_log    /var/log/cloudify/nginx/cloudify-files.log;
-  location / {
-    root              /opt/manager/resources;
-    autoindex on;
-    allow             all;
-    deny              all;
-  }
-}
+include "/etc/nginx/conf.d/{{ ctx.source.instance.runtime_properties.rest_protocol }}-file-server.cloudify";
```
Comment (dankilman) [inline components/nginx/config/default.conf]: check if required

File: components/nginx/config/fileserver-location.cloudify [hunk 7]
```diff
@@ -1,4 +1,4 @@
-  location ~ ^/resources {
+location ~ ^/resources {
```
Comment (dankilman) [inline components/nginx/config/fileserver-location.cloudify]: rest redirect

File: components/nginx/config/https-file-server.cloudify [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# internal file server
+server {
```
Comment (tehasdf) [inline components/nginx/config/https-file-server.cloudify]: this needs a `include fileserver-location.cloudify` i think?

File: components/nginx/config/default.conf (hunk 6 above)
Comment (noakup) [inline components/nginx/config/default.conf]: required..

File: components/nginx/config/https-rest-server.cloudify [hunk 9]
```diff
@@ -23,16 +23,16 @@ server {
...
 
-# REST and UI external server.
+# REST and UI external server
 server {
-  # server listening for the external requests.
+  # server listening for external requests
   # listens on both 443 and 80, but http will be redirected to https (below)
   listen              80;
-  listen              443 ssl;
+  listen              443 ssl default_server;
```
Comment (dankilman) [inline components/nginx/config/https-rest-server.cloudify]: what does this change mean?

**Pull request merged**